iterative/utils/analytics.go

```diff
@@ -0,0 +1,351 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+	"github.com/wessie/appdirs"
+
+	"golang.org/x/crypto/scrypt"
+)
+
+const (
+	Timeout  = 5 * time.Second
+	Endpoint = "https://telemetry.cml.dev/api/v1/s2s/event?ip_policy=strict"
+	Token    = "s2s.jtyjusrpsww4k9b76rrjri.bl62fbzrb7nd9n6vn5bpqt"
+)
+
+var (
+	Version string = "0.0.0"
+	wg      sync.WaitGroup
+)
+
+func getenv(key, defaultValue string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return defaultValue
+	}
+	return value
+}
+
+func deterministic(data string) (*uuid.UUID, error) {
+	ns := uuid.NewSHA1(uuid.NameSpaceDNS, []byte("iterative.ai"))
+
+	seed, err := ns.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	dk, err := scrypt.Key([]byte(data), seed, 1<<16, 8, 1, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	id := uuid.NewSHA1(ns, []byte(hex.EncodeToString(dk)))
+	return &id, nil
+}
+
+func SystemInfo() map[string]interface{} {
+	hostStat, _ := host.Info()
+	return map[string]interface{}{
+		"os_name":          hostStat.OS,
+		"platform_version": hostStat.PlatformVersion,
+	}
+}
+
+func TaskDuration(logs string) float64 {
+	regex := regexp.MustCompile(`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}`)
+	matches := regex.FindAllString(logs, -1)
+	taskDuration := 0.0
+
+	if len(matches) > 1 {
+		layout := "2006-01-02 15:04:05"
+		t1, _ := time.Parse(layout, matches[len(matches)-1])
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
+	}
+
+	return taskDuration
+}
+
+func IsCI() bool {
+	if _, ok := os.LookupEnv("CI"); ok {
+		return true
+	}
+
+	return len(guessCI()) > 0
+}
+
+func guessCI() string {
+	if _, ok := os.LookupEnv("GITHUB_SERVER_URL"); ok {
+		return "github"
+	}
+
+	if _, ok := os.LookupEnv("CI_SERVER_URL"); ok {
+		return "gitlab"
+	}
+
+	if _, ok := os.LookupEnv("BITBUCKET_WORKSPACE"); ok {
+		return "bitbucket"
+	}
+
+	if _, ok := os.LookupEnv("TF_BUILD"); ok {
+		return "azure"
+	}
```

Comment:
in light of https://github.com/iterative/terraform-provider-iterative/pull/493/files#r906232930

```suggestion
	return len(guessCI()) > 0
}

func guessCI() string {
	if _, ok := os.LookupEnv("GITHUB_SERVER_URL"); ok {
		return "github"
	}

	if _, ok := os.LookupEnv("CI_SERVER_URL"); ok {
		return "gitlab"
	}

	if _, ok := os.LookupEnv("BITBUCKET_WORKSPACE"); ok {
		return "bitbucket"
	}

	if _, ok := os.LookupEnv("TF_BUILD"); ok {
		return "azure"
	}

	if _, ok := os.LookupEnv("CI"); ok {
		return "unknown"
	}
```

Reply:
With the addition of 
```go
if _, ok := os.LookupEnv("CI"); ok {
		return "unknown"
	}
```

I guess we should refactor isCI as 

```go
func IsCI() bool {
	return len(guessCI()) > 0 
}
```